heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/service/AccessTokenInterceptorService.java

```diff
@@ -87,34 +88,29 @@ private void validateAccessToken(Long apiId, String clientId, String accessToken
 
         TraceContextHolder.getInstance().getActualTrace().setAccessToken(DigestUtils.digestMD5(accessToken));
 
-        if (clientId == null || clientId.isEmpty()) {
+        if (Objects.isNull(clientId) || clientId.isEmpty()) {
             buildResponse(String.format(ConstantsInterceptors.GLOBAL_CLIENT_ID_OR_ACESS_TOKEN_NOT_FOUND, "Client Id"));
             return;
         }
 
-        if (accessToken != null && !accessToken.isEmpty()) {
+        if (Objects.nonNull(accessToken) &&
+                !accessToken.isEmpty() &&
+                Objects.nonNull(accessTokenRepository.findAccessTokenActive(accessToken)) &&
```

Comment:
Keeping this way, you are doing the same search three times! I think the previous version of this file makes more sense